src/modules/ipc/ipc.c

```diff
@@ -1,7 +1,226 @@
 #include <kernel/ipc.h>
 #include "private/_ipc.h"
+#include <stdlib.h> // For memory allocation
+#include <string.h> // For memset
+#include <time.h>   // For clock_gettime
 
-void ipc_init(void) {
-    
-    (void) sizeof(struct ipc_internal);
+#define MAX_ENDPOINTS 128
+
+static struct ipc_endpoint endpoint_table[MAX_ENDPOINTS];
+
+/* Retrieve CPU time in milliseconds */
+static uint32_t kernel_get_time_ms(void)
```

Comment:
```suggestion
static inline uint32_t kernel_get_time_ms(void)
```

It should also be placed in a util.h file to make sure definitions are clear.